Refactoring pipeline log tests

16yuki0702 opened the pull request:
<!-- 🎉🎉🎉 Thank you for the PR!!! 🎉🎉🎉 -->

# Changes
This is a first step of https://github.com/tektoncd/cli/issues/520

next is refactoring of pipeline start tests.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Regenerate the manpages and docs with `make docs` and `make man` if needed.
- [ ] Run the code checkers with `make check`
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

<!--
Does your PR contain User facing changes?

If so, briefly describe them here so we can include this description in the
release notes for the next release!
-->

```
release-note
```

piyush-garg commented:
/ok-to-test

danielhelfand commented on pkg/cmd/pipeline/logs_test.go:
```diff
@@ -239,270 +127,97 @@ func TestLogs_interactive_get_all_inputs(t *testing.T) {
-					cb.PipelineRunCompletionTime(clock.Now().Add(10*time.Minute)),
-				),
-			),
+			name:      "Invalid name space",
```
```suggestion
			name:      "Invalid namespace",
```

danielhelfand commented on pkg/cmd/pipeline/logs_test.go:
```diff
@@ -239,270 +127,97 @@ func TestLogs_interactive_get_all_inputs(t *testing.T) {
-				cb.PipelineCreationTimestamp(clock.Now().Add(-15*time.Minute)),
-			),
+		{
+			name:      "Specify not exist pipeline",
```
```suggestion
			name:      "Pipeline does not exist",
```

danielhelfand commented on pkg/cmd/pipeline/logs_test.go:
```diff
@@ -239,270 +127,97 @@ func TestLogs_interactive_get_all_inputs(t *testing.T) {
-				),
-			),
+		{
+			name:      "Specify not exist pipelinerun",
```
```suggestion
			name:      "Pipelinerun does not exist",
```

danielhelfand commented on pkg/cmd/pipeline/logs_test.go:
```diff
@@ -239,270 +127,97 @@ func TestLogs_interactive_get_all_inputs(t *testing.T) {
-				},
-			},
+		{
+			name:      "Specify invalid limit number",
```
```suggestion
			name:      "Invalid limit number",
```

danielhelfand commented on pkg/cmd/pipeline/logs_test.go:
```diff
@@ -239,270 +127,97 @@ func TestLogs_interactive_get_all_inputs(t *testing.T) {
 }
 
-func TestLogs_interactive_limit_1(t *testing.T) {
+func TestPipelineLog2(t *testing.T) {
```
```suggestion
func TestPipelineLog_Interactive(t *testing.T) {
```

Let's have a more descriptive name. It looks like this series of tests is about testing the interactive aspects of pipeline logs. Feel free to suggest something more apt.

16yuki0702 replied:
Thanks!! I totally agree and modified the name :thumbsup:

danielhelfand commented on pkg/cmd/pipeline/logs_test.go:
```diff
@@ -648,264 +303,294 @@ func TestLogs_interactive_ask_all_last_run(t *testing.T) {
-				}
-
-				return nil
+			name:      "basic interaction1",
```
```suggestion
			name:      "Select output-pipeline-run-2 logs",
```

danielhelfand commented on pkg/cmd/pipeline/logs_test.go:
```diff
@@ -648,264 +303,294 @@ func TestLogs_interactive_ask_all_last_run(t *testing.T) {
-				ObjectMeta: metav1.ObjectMeta{
-					Name: ns,
+		{
+			name:      "basic interaction2",
```
```suggestion
			name:      "Pipeline included with command and select pipelinerun output-pipeline-run",
```

danielhelfand commented on pkg/cmd/pipeline/logs_test.go:
```diff
@@ -648,264 +303,294 @@ func TestLogs_interactive_ask_all_last_run(t *testing.T) {
+
+					if _, err := c.SendLine(string(terminal.KeyEnter)); err != nil {
+						return err
+					}
```
Why in this case are we expecting a pipeline prompt if the pipeline name is included as part of the command?

16yuki0702 replied:
Sorry I don't know why this test case expect it :sweat_smile:
(because I just transferred original test cases to table driven tests)

but actually it seems no need expect it, so I removed these :thumbsup:

danielhelfand commented on pkg/cmd/pipeline/logs_test.go:
```diff
@@ -648,264 +303,294 @@ func TestLogs_interactive_ask_all_last_run(t *testing.T) {
-				}
-
-				return nil
+			name:      "basic interaction3",
```
```suggestion
			name:      "Set pipelinerun limit=2 and select pipelinerun output-pipeline-run",
```

danielhelfand commented on pkg/cmd/pipeline/logs_test.go:
```diff
@@ -648,264 +303,294 @@ func TestLogs_interactive_ask_all_last_run(t *testing.T) {
+
+					if _, err := c.SendLine(string(terminal.KeyEnter)); err != nil {
+						return err
+					}
```
Why in this case are we expecting a pipeline prompt if the pipeline name is included as part of the command?

danielhelfand commented on pkg/cmd/pipeline/logs_test.go:
```diff
@@ -648,264 +303,294 @@ func TestLogs_interactive_ask_all_last_run(t *testing.T) {
-				ObjectMeta: metav1.ObjectMeta{
-					Name: ns,
+		{
+			name:      "basic interaction4",
```
```suggestion
			name:      "Set pipelinerun limit=1 and select pipelinerun output-pipeline-run-2",
```

danielhelfand commented on pkg/cmd/pipeline/logs_test.go:
```diff
@@ -648,264 +303,294 @@ func TestLogs_interactive_ask_all_last_run(t *testing.T) {
-				),
-			),
+		{
+			name:      "basic interaction5",
```
```suggestion
			name:      "Select pipeline output-pipeline with last=true ",
```

danielhelfand commented on pkg/cmd/pipeline/logs_test.go:
```diff
@@ -648,264 +303,294 @@ func TestLogs_interactive_ask_all_last_run(t *testing.T) {
+
+					if _, err := c.ExpectEOF(); err != nil {
+						return err
+					}
```
I'm a bit confused about this aspect of the test since `last` is true.

danielhelfand commented on pkg/cmd/pipeline/logs_test.go:
```diff
@@ -648,264 +303,294 @@ func TestLogs_interactive_ask_all_last_run(t *testing.T) {
-				ObjectMeta: metav1.ObjectMeta{
-					Name: ns,
+		{
+			name:      "basic interaction6",
```
```suggestion
			name:      "Include pipeline output-pipeline as part of command",
```

danielhelfand commented on pkg/cmd/pipeline/logs_test.go:
```diff
@@ -648,264 +303,294 @@ func TestLogs_interactive_ask_all_last_run(t *testing.T) {
+			limit:     5,
+			last:      false,
+			namespace: ns,
+			input:     cs2,
```
`cs2` seems to be the exact same as `cs`. Is it needed?

16yuki0702 replied:
the difference between cs and cs2 is namespace.
(cs uses "ns", cs2 uses ns)

so cs2 is used for another namespace test case!
(I agree it cause a bit confuse :sweat_smile:)

danielhelfand replied:
Yes, good catch! I should've caught this as I added this 😄

danielhelfand commented on pkg/cmd/pipeline/logs_test.go:
```diff
@@ -648,264 +303,294 @@ func TestLogs_interactive_ask_all_last_run(t *testing.T) {
-				}
-
-				return nil
+			name:      "basic interaction7",
```
```suggestion
			name:      "Select pipeline output-pipeline",
```

vdemeester commented on pkg/cmd/pipeline/logs_test.go:
```diff
@@ -16,15 +16,16 @@ package pipeline
 	"github.com/jonboulle/clockwork"
+	"github.com/tektoncd/cli/pkg/cli"
 	"github.com/tektoncd/cli/pkg/helper/options"
+	htest "github.com/tektoncd/cli/pkg/helper/test"
```
I feel this package would be better at `github.com/tektoncd/cli/test/prompt` (for the prompt part). It's clearer what it does : it's under test, so it's a test helper, and it's `prompt`, so it's prompt-related functions… *and* it wouldn't clash with `github.com/tektoncd/cli/test` package :angel:

16yuki0702 replied:
Thanks!! addressed 👍

16yuki0702 commented:
@danielhelfand @vdemeester Thank for reviews and sorry for late :pray:
I addressed the reviews, but when I squashed the pr this pr was closed for some reason and couldn't reopen :innocent:

I'll open another PR :innocent:

16yuki0702 commented:
@danielhelfand @vdemeester 
>  I addressed the reviews, but when I squashed the pr this pr was closed for some reason and couldn't reopen 😇

Oops... I just missed the operation 😇
sorry, I reopened :pray:

16yuki0702 commented on test/prompt/prompt.go:
```diff
@@ -25,11 +25,14 @@ import (
 )
 
+// PromptTest provides test utility for prompt test.
 type PromptTest struct {
```
Mmm... Should I write this test case for golint errors 🤔??

danielhelfand replied:
```
I1224 05:03:26.489] test/prompt/prompt.go:29:6: type name will be used as prompt.PromptTest by other packages, and that stutters; consider calling this Test (golint)
I1224 05:03:26.490] type PromptTest struct {
```

So I think it wants you to remove `Test` from the struct name.

16yuki0702 replied:
Ahh... I totally understood, thanks!!
fixed the name :thumbsup:

danielhelfand commented:
/test pull-tekton-cli-unit-tests

16yuki0702 commented:
/test pull-tekton-cli-unit-tests

danielhelfand commented:
/test pull-tekton-cli-unit-tests

danielhelfand commented:
/lgtm
